offchain-modules/packages/x/src/handlers/eth.ts

```diff
@@ -148,12 +163,115 @@ export class EthHandler {
 
   async handleLog(log: Log, currentHeight: number): Promise<void> {
     const parsedLog = await this.ethChain.iface.parseLog(log);
-    if (parsedLog.name === 'Locked') {
-      await this.onLockLogs(log, parsedLog, currentHeight);
-    } else if (parsedLog.name === 'Unlocked') {
-      await this.onUnlockLogs(log, parsedLog);
+    switch (parsedLog.name) {
+      case 'Locked':
+        await this.onLockLogs(log, parsedLog, currentHeight);
+        break;
+      case 'Unlocked':
+        await this.onUnlockLogs(log, parsedLog);
+        break;
+      case 'Burn':
+        await this.onBurnLogs(log, parsedLog, currentHeight);
+        break;
+      case 'Mint':
+        await this.onMinted(log, parsedLog);
+        break;
+      default:
+        logger.info(`not handled log type ${parsedLog.name}, log: ${JSON.stringify(log)}`);
+    }
+  }
+
+  async onMinted(log: Log, parsedLog: ParsedLog): Promise<void> {
```

Comment:
The logic of `onMinted` is incorrect. Not only collector, watcher and verifier also run `onMinted`. Please read and understand `onUnlockLogs`, the main logic is similar to `onMinted`.

https://github.com/nervosnetwork/force-bridge/blob/a85d9b8e4ea4e4d55b55f2f86e22380e496e1f04/offchain-modules/packages/x/src/handlers/eth.ts#L249

Reply:
In my opinion, when we receive the data from Ethereum chain, but we can't find the record in the collect mint table, this must be incorrect, we should do nothing but to log the transaction information, right?

Reply:
Finally, I understand, that there are three roles, watcher, validator and collector, they are deployed to different server and are independence from each other, include the db and any other services.